### ISSUE TITLE
LOCALE_PATH multiple printing

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#1687]
`LOCALE_PATH` is printed to `stderr` by every MPI process from line 210 here.
https://github.com/boutproject/BOUT-dev/blob/6e1d4004646d1c35eb4ca129a1f8268c943555cc/src/bout%2B%2B.cxx#L198-L213

L210 is printing when gettext stuff is set correctly, I think this should be going to stdout rather than stderr.

Is it possible to print it from somewhere else, so that the information is only printed once per process? Maybe during initialization of output?

**Comment by dschwoerer** [comment]
Does it need to be printed at all by default?
It is quite useful for debugging, but I think it should be disabled by
default ...

**Comment by bendudson** [comment]
Probably does not need to be printed, though something in the log files would be useful. Unfortunately it's too early at that point to use `output`. Perhaps the `BOUT_LOCALE_PATH` could be printed to `output` later, along with the other compile-time options.

**Comment by johnomotani** [comment]
Just realised from @bendudson's comment that `BOUT_LOCALE_PATH` is already printed with the compile-flags, e.g. there's a line in my log files like
`
Compiled with flags : "-DMD5SUM=f2a1e40b894df85756fd32e5162cdd94 -DREVISION=6e1d4004646d1c35eb4ca129a1f8268c943555cc -g -Wall -Wextra -Wnull-dereference -O3 -march=native -funroll-loops -DSIGHANDLE -DLOGCOLOR -DBOUT_VERSION_STRING=\"4.2.2\" -DBOUT_VERSION_DOUBLE=4.22 -DBACKTRACE -fPIC -DBOUT_HAS_FFTW -DNCDF4 -DHDF5 -DLAPACK -DBOUT_HAS_IDA -DIDAINT=long -DBOUT_HAS_CVODE -DCVODEINT=long -DBOUT_HAS_ARKODE -DARKODEINT=long -DBOUT_HAS_PVODE -DBOUT_HAS_GETTEXT -DBOUT_LOCALE_PATH=/usr/local/share/locale"
`

Seems like we can just remove L210. I'll make a PR...

**Comment by johnomotani** [comment]
PS I'm having difficulty understanding the error message
https://github.com/boutproject/BOUT-dev/blob/6e1d4004646d1c35eb4ca129a1f8268c943555cc/src/bout%2B%2B.cxx#L212
Should we include the locale path here, like `...Could not set locale=<BOUT_LOCALE_PATH>...`? Would you try a "`different LANG setting`" at compile time (-> `Try compiling with a different LANG setting.`) or with an environment variable (-> `Try setting a different environment variable $LANG.`) or...?

**Comment by d7919** [comment]
> PS I'm having difficulty understanding the error message
> [BOUT-dev/src/bout++.cxx](https://github.com/boutproject/BOUT-dev/blob/6e1d4004646d1c35eb4ca129a1f8268c943555cc/src/bout%2B%2B.cxx#L212)
> 
> Line 212 in [6e1d400](/boutproject/BOUT-dev/commit/6e1d4004646d1c35eb4ca129a1f8268c943555cc)
> 
>  fprintf(stderr, "WARNING: Could not set locale. Try a different LANG setting\n"); 
> 
> Should we include the locale path here, like `...Could not set locale=<BOUT_LOCALE_PATH>...`? Would you try a "`different LANG setting`" at compile time (-> `Try compiling with a different LANG setting.`) or with an environment variable (-> `Try setting a different environment variable $LANG.`) or...?

I think this is a runtime setting taken from the environment so the message could be changed to note this -- but I think there is possibly another potential issue at configure time of an incorrect `BOUT_LOCALE_PATH` being set. I suspect that this will be caught by configure and an appropriate error message given there so it should probably only arise if the user has modified the value of `BOUT_LOCALE_PATH` by hand.

**Comment by johnomotani** [comment]
How about a message like:
```
"WARNING: Could not set locale. Check the LANG environment variable (get available values by running 'locale -a'). If LANG is correct, there may be a problem with the BOUT_LOCALE_PATH that BOUT++ was compiled with."
```
? Additions/improvements welcome :smiley:

**Comment by dschwoerer** [comment]
Resolved by #1689
